ppr-ui/src/components/common/PartyForm.vue

```diff
@@ -0,0 +1,175 @@
+          :orgNameRules="schema.businessName"
+          @updateOrgName="partyModel.businessName = $event"
+        />
+    </article>
+
+    <!-- Email Address -->
+    <article v-if="hasPropData('emailAddress')" class="mt-3">
+      <label class="generic-label" for="contact-info-email">Email Address</label>
+      <v-text-field
+        filled
+        id="contact-info-email"
+        class="pt-4 pr-2"
+        label="Email Address (Optional)"
+        v-model="partyModel.emailAddress"
+        :rules="schema.email"
+      />
+    </article>
+
+    <!-- Phone Number -->
+    <article v-if="hasPropData('phoneNumber')" class="mt-3">
+      <label class="generic-label" for="contact-info-phone">Phone Number</label>
+
+      <v-row no-gutters class="mt-5">
+        <v-col>
+          <v-text-field
+              v-mask="'(NNN) NNN-NNNN'"
+              filled
+              id="party-form-phone"
+              class="pr-3"
+              label="Phone Number"
+              v-model="partyModel.phoneNumber"
+              :rules="schema.phone"
+          />
+        </v-col>
+        <v-col>
+          <v-text-field
+              filled
+              id="party-form-phone-ext"
+              class="px-2"
+              label="Extension (Optional)"
+              v-model="partyModel.phoneExtension"
+              :rules="schema.phoneExt"
+          />
+        </v-col>
+      </v-row>
+    </article>
+
+    <!-- Mailing Address -->
+    <article v-if="hasPropData('address')" class="mt-3">
+      <label class="generic-label" for="party-form-address">Mailing Address</label>
+
+      <BaseAddress
+        editing
+        class="mt-5"
+        id="party-form-address"
+        ref="baseAddressRef"
+        :schema="schema.address"
+        :value="partyModel.address"
+        @valid="isAddressValid = $event"
+      />
+    </article>
+  </v-form>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, reactive, ref, toRefs, watch } from 'vue-demi'
+import { FormIF, OrgLookupConfigIF, PartyIF, PartySchemaIF } from '@/interfaces'
+import { BaseAddress } from '@/composables/address'
+import { VueMaskDirective } from 'v-mask'
+import OrgNameLookup from '@/components/common/OrgNameLookup.vue'
+
+export default defineComponent({
+  name: 'PartyForm',
+  emits: ['isValid'],
+  components: {
+    BaseAddress,
+    OrgNameLookup
+  },
+  props: {
+    baseParty: {
+      type: Object as () => PartyIF,
+      required: true
+    },
+    schema: {
+      type: Object as () => PartySchemaIF,
+      default: null
+    },
+    orgLookupConfig: {
+      type: Object as () => OrgLookupConfigIF,
+      default: null
+    }
+  },
+  directives: {
+    mask: VueMaskDirective
+  },
+  setup (props, { emit }) {
+    const partyFormRef = ref(null) as FormIF
+    const baseAddressRef = ref(null) as FormIF
+
+    const localState = reactive({
+      isFormValid: false,
+      isAddressValid: false,
+      partyModel: props.baseParty as PartyIF,
```

Comment:
Note i don't emit and set every single property captured in the form. 
All we need to do is pass the reference to store into the form, as the form model updates, it's updated in our state.

Very little handling and also helps decouple the component from specific store values. 
You can pass whatever store model you have in as the baseParty and as long as it's initialized there it will be reactive.

Reply:
That's awesome lots of unnecessary emitting then, much cleaner!